temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java

```diff
@@ -355,8 +355,12 @@ public void handle(WorkflowTask task) throws Exception {
 
           if (!locked) {
             throw new UnableToAcquireLockException(
-                "Workflow lock for the run id hasn't been released by one of previous execution attempts, "
-                    + "consider increasing workflow task timeout.");
+                "Workflow lock for the run id hasn't been released by one of previous execution attempts.  "
+                    + "This likely means the workflow task timed out, and the Service attempted to "
```

Comment:
This exception can be thrown any time the workflow lock cannot be acquire because another task is holding the lock. It can happen because of workflow task timeout, but that is misleading to suggest it is likely or to recommend investigating Workflow Task timed out without doing some triage. This error can have multiple causes including something happening on the Temporal Server side

```suggestion
"Workflow lock for the run id hasn't been released by one of previous execution attempts. This means another workflow task is holding the lock. If seen repeatedly users should investigate the failing workflow executions for more details."
```